### PR TITLE
Cosmos DB Terraform Module

### PR DESCRIPTION
**Pull request by ericpera-msft**
As a developer, I can now configure and deploy a Cosmos DB instance

## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [NA] I have added tests to cover my changes.
* [NA] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
- The ability to create and deploy an instance of Cosmos DB does not exist.

## What is the new behavior?
-------------------------------------
- Users can now use terraform to configure and deploy an instance of Cosmos DB.

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: infra/modules/providers/azure/cosmosdb/output.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+output "cosmosdb_endpoint" {
+  description = "The endpoint used to connect to the CosmosDB account."
+  value       = "${azurerm_cosmosdb_account.cosmosdb.endpoint}"
+}
+
+output "cosmosdb_primary_master_key" {
+    description = "The Primary master key for the CosmosDB Account."
+    value = "${azurerm_cosmosdb_account.cosmosdb.primary_master_key}"
```
Comment (erikschlegel) [inline infra/modules/providers/azure/cosmosdb/output.tf]: this output should be marked as `sensitive = true`

Comment (ericpera-msft) [reply infra/modules/providers/azure/cosmosdb/output.tf]: @erikschlegel Done, pushed the updated output to the PR.

**Comment by frodopwns** [comment]
Looks like we have a failing test. Is this something we can easily run locally? If not, can we get access to the logs, or can someone paste the issue?

**Comment by ericpera-msft** [comment]
@erikschlegel If you get a second, can you send us the logs from the build test that's failing in the pipeline?

**Comment by erikschlegel** [comment]
> @erikschlegel If you get a second, can you send us the logs from the build test that's failing in the pipeline?

Apologies for the confusion here @ericpera-msft . I believe @nmiodice is currently investigating this build failure which seems to be unrelated to the changes in your PR. @nmiodice will confirm.

**Comment by nmiodice** [comment]
Hey I checked this out. The root cause was related to a stale deployment we had. Were still learning with Terraform, forgive us :)

I re-ran your build and it checks out

File: infra/modules/providers/azure/cosmosdb/output.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+output "cosmosdb_endpoint" {
+  description = "The endpoint used to connect to the CosmosDB account."
+  value       = "${azurerm_cosmosdb_account.cosmosdb.endpoint}"
+}
+
+output "cosmosdb_primary_master_key" {
```
Comment (erikschlegel) [inline infra/modules/providers/azure/cosmosdb/output.tf]: You may want to consider adding the `connection_strings` here as well?

Comment (ericpera-msft) [reply infra/modules/providers/azure/cosmosdb/output.tf]: Will do, pushing that change in the next 20 minutes

**Comment by ericpera-msft** [comment]
@nmiodice No problem :) Thanks for looking into it!

**Pull request merged**